index.js

```diff
@@ -18,12 +18,18 @@ function findAllAfter(parent, index, test) {
   children = parent.children
   length = children.length
 
-  if (index && index.type) {
+  if (index === undefined) {
+    throw new Error('Expected positive finite index or child node')
+  } else if (index && typeof index !== 'number') {
     index = children.indexOf(index)
+    if (index < 0) {
```

Comment:
```suggestion
    if (index === -1) {
```

`index < 0` can only be one value, this is more explicit.